geomacy/brooklyn-dist
# Do not merge - Test PR 72

Test files for https://github.com/apache/brooklyn-dist/pull/72.  

Contains a class that gets renamed from VacantSoftwareProcess in version 0.1.0 to VoidSoftwareProcess in 0.2.0; test is to see if I can use the user-specific renames file to get Karaf to rebind with the updated bundle.

test procedure:
- Build the test bundle at version 0.1.0 (VacantSoftwareProcess)
- bundle:install mvn:org.apache.brooklyn/temp-delete-me/0.1.0 and start the bundle
- Deploy the following blueprint
```yaml
name: test
location: amazon-eu-west-1
services:
- type: org.apache.brooklyn.temp-delete-me:org.apache.brooklyn.geofftest.VacantSoftwareProcess
  brooklyn.config:
    some.random.config: wow
```
- Build the test bundle at version 0.2.0 (VoidSoftwareProcess)
- Extract a fresh Karaf
- Start it (noticing rebind failures in the logs)
-  bundle:install mvn:org.apache.brooklyn/temp-delete-me/0.2.0 and start it.
- stop Karaf 
- edit the `etc/org.apache.brooklyn.classrename.cfg` file to specify:
```yaml
org.apache.brooklyn.geofftest.VacantSoftwareProcess=org.apache.brooklyn.geofftest.VoidSoftwareProcess
org.apache.brooklyn.geofftest.VacantSoftwareProcessImpl=org.apache.brooklyn.geofftest.VoidSoftwareProcessImpl
```
- start Karaf again
- check that rebind succeeds this time
